### ISSUE TITLE
Display randomly jumping while printing

### ISSUE THREAD

**Issue by josuah** [brilliantlabsAR/monocle-micropython#208]
Using this snippet, the display should keep printing random numbers, but also shifts the vertical and horizontal in a "pacman-repeat: fashion:

```python
import display
from random import random

while True:
    l = []
    y = 0
    for i in range(8):
        l.append(display.Text(f'{random()}{random()}{random()}', 0, y, display.WHITE))
        y += 50

    display.brightness(0)
    display.show(l)
```

![Image](https://user-images.githubusercontent.com/55351021/240059528-369872bb-a502-4397-976d-2496de081850.gif)

**Comment by josuah** [comment]
On the other hand, printing constantly the same data does not seems to trigger any glitch:

```python
from display import *

while True:
    show([
        Text('0.54864070.42984010.3100815', 0, 0, 0xffffff),
        Text('0.35902540.63464520.06198609', 0, 50, 0xffffff),
        Text('0.57318440.86822380.2571942', 0, 100, 0xffffff),
        Text('0.43299210.42563390.09295154', 0, 150, 0xffffff),
        Text('0.067244410.34240930.9991117', 0, 200, 0xffffff),
        Text('0.68130370.072521210.079427', 0, 250, 0xffffff),
        Text('0.79938940.8310890.2462578', 0, 300, 0xffffff),
        Text('0.52144990.59871090.1736188', 0, 350, 0xffffff),
    ])
```

**Comment by josuah** [comment]
A shorter reproducer showing a single line of text:
```python
from display import *

while True:
    s = f'{random()}{random()}{random()}'
    show(Text(s, 0, 0, 0xffffff))
```

**Comment by josuah** [comment]
By inserting delays between prints, there is less often any glitch happening, sometimes very rarely.

This snippet did produce the glitch occasionally:

```python
from display import *
from random import random
from time import sleep_ms

while True:
    s = f'{random()}{random()}{random()}'
    show(Text(s, 0, 0, 0xffffff))
    print(s)
    sleep_ms(50)
```

But by replaying the text that did produce a bug, the offset glitch does not show-up anymore:

```python
from display import *
from random import random
from time import sleep_ms

lines = [
    '0.5805330.15940280.6911397',
    '0.79491080.43136680.3676161',
    '0.63313060.0084255930.9829068',
    '0.63552930.8199680.9345885',
    '0.59016860.86261960.8806037',
    '0.34546230.33397380.03462267',
    '0.52931680.4543540.4723445',
    '0.39820990.90908970.5736155',
    '0.012948750.20397440.8955362',
    '0.59366110.9991580.764183',
    '0.93191710.23650120.189626',
    '0.089686760.93767630.04712594',
    '0.8846930.91967950.3160187',
    '0.038520930.65211540.3805225',
    '0.8610210.12769660.08911633',
    '0.14713990.803730.5070182',
    '0.52613640.36443480.9711175',
    '0.0065060850.54428820.1900092',
    '0.33138780.006158710.1245269',
    '0.1975520.010668280.6214256',
    '0.53798060.32947040.3933594',
    '0.589480.043236850.9271793',
    '0.012523890.78625310.6820898',
    '0.96446270.2030720.3585348',
    '0.32869330.64580760.3379695',
    '0.94329910.324850.5810333',
    '0.35280440.21398450.4394406',
    '0.29259660.70331120.5737033',
    '0.11312340.48636320.05542839',
    '0.69707190.2586130.007415295',
    '0.11193860.68077210.931271',
    '0.37418390.31486460.335043',
    '0.51688080.26435510.2860227',
    '0.7012670.17604420.9427845',
    '0.93389740.41557050.8782581',
    '0.95420190.83162290.1003851',
    '0.60233410.60269330.5109658',
    '0.35414360.24549960.6465296',
    '0.65653680.94330510.4313878',
    '0.25421420.92393830.8194826',
    '0.52898790.35168420.1732073',
]

for s in lines:
    show(Text(s, 0, 0, 0xffffff))
    print(s)
    sleep_ms(50)
```

This suggests that it is not related to the content, but instead by the moment where the glitch happens.

Maybe printing text at the exact same moment it is being flushed to the display?

What happens to the SRAM read address while it is both read and written at the same time?
Should there be some locking mechanism here?

**Comment by josuah** [comment]
Cross reference: https://github.com/sathibault/vgrs/issues/13

**Comment by josuah** [comment]
The approach proposed by @sathibault to fix it is to use double-buffering to avoid clashing a same memory region.

**Comment by josuah** [comment]
Fixed by #210
